server/methods.js

```diff
@@ -164,17 +170,34 @@ function createProductImage(product) {
   });
 }
 
+async function createProductImageFromUrl(product) {
+  const url = Promise.await(randomPuppy());
+  const fileRecord = await FileRecord.fromUrl(url, { fetch });
+  const topVariant = getTopVariant(product._id);
+  const { shopId } = product;
+  fileRecord.metadata = {
+    productId: product._id,
+    variantId: topVariant._id,
+    toGrid: 1,
+    shopId,
+    priority: 0,
+    workflow: "published"
+  };
+  return Media.insert(fileRecord);
+}
+
 /**
  * @method attachProductImages
  * @summary Generate an image and attach it to every product
  * @returns {undefined}
  */
 function attachProductImages() {
   Logger.info("Started loading product images");
-  const products = Products.find({ type: "simple" }).fetch();
+  const products = Products.find({type: "simple"}).fetch();
   for (const product of products) {
-    if (!Promise.await(Media.findOne({ "metadata.productId": product._id }))) {
-      createProductImage(product);
+    if (!Promise.await(Media.findOne({"metadata.productId": product._id}))) {
```

Comment:
This will be a bit faster if you use `MediaRecords` rather than `Media`. `MediaRecords` is the pure collection importable from `"/lib/collections"` just like all the others. It won't do the extra step of converting the document to a FileRecord instance. Since it's a normal collection, you also won't need the `Promise.await` around it.

That said, an even better approach would be doing a single fetch, if it doesn't take up too much memory.

```js
const productIds = products.map(({ _id }) => _id);
const media = MediaRecords.find({ "metadata.productId": { $in: productIds } }).fetch();
const productIdsWithMedia = _.uniq(media.map((doc) => doc.metadata.productId));
// Then loop through `products` and check whether each is in `productIdsWithMedia`. If not, make image.
```